### PR TITLE
Added gobo presets in Clay-Paky, Contest, DTS, Dune_Ligting, Elation, Elumen8

### PR DESCRIPTION
**Pull request by kluszon**
Added gobo presets in Clay-Paky, Contest, DTS, Dune_Ligting, Elation, Elumen8

### CODE REVIEW

File: resources/fixtures/Contest/Contest-PSC-930.qxf [hunk 1]
```diff
@@ -39,14 +39,14 @@
...
+  <Capability Min="190" Max="255" Preset="RotationClockwise">Gobo rainbow</Capability>
  </Channel>
  <Channel Name="Gobo rotation">
   <Group Byte="0">Speed</Group>
-  <Capability Min="0" Max="127">Index</Capability>
-  <Capability Min="128" Max="187">Clockwise</Capability>
-  <Capability Min="188" Max="193">Stationary</Capability>
-  <Capability Min="194" Max="255">Counter Clockwise</Capability>
+  <Capability Min="0" Max="127" Preset="RotationIndexed">Index</Capability>
+  <Capability Min="128" Max="187" Preset="RotationClockwiseSlowToFast">Clockwise</Capability>
```
Comment (lady-darkness) [inline resources/fixtures/Contest/Contest-PSC-930.qxf:47]: without speed?

File: resources/fixtures/Contest/Contest-PSC-930.qxf [hunk 2]
```diff
@@ -39,14 +39,14 @@
...
+  <Capability Min="188" Max="193" Preset="RotationStop">Stationary</Capability>
+  <Capability Min="194" Max="255" Preset="RotationCounterClockwiseSlowToFast">Counter Clockwise</Capability>
```
Comment (lady-darkness) [inline resources/fixtures/Contest/Contest-PSC-930.qxf:49]: without speed?

File: resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Beam-1500.qxf [hunk 3]
```diff
@@ -64,46 +64,40 @@
...
-  <Capability Min="184" Max="195">Gobo 3 Shake Slow -&gt; Fast</Capability>
-  <Capability Min="196" Max="207">Gobo 4 Shake Slow -&gt; Fast</Capability>
-  <Capability Min="208" Max="219">Gobo 5 Shake Slow -&gt; Fast</Capability>
-  <Capability Min="220" Max="231">Gobo 6 Shake Slow -&gt; Fast</Capability>
-  <Capability Min="232" Max="243">Gobo 7 Shake Slow -&gt; Fast</Capability>
-  <Capability Min="244" Max="255">Gobo 8 Shake Slow -&gt; Fast</Capability>
+  <Capability Min="72" Max="113" Preset="RotationClockwiseFastToSlow">Fast -&gt; Slow Rotation</Capability>
+  <Capability Min="114" Max="117" Preset="RotationStop">Stop</Capability>
+  <Capability Min="118" Max="159" Preset="RotationCounterClockwiseSlowToFast">Slow -&gt; Fast Rotation</Capability>
+  <Capability Min="160" Max="171" Preset="GoboShakeMacro" Res1="">Gobo 1 Shake Slow -&gt; Fast</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Beam-1500.qxf:70]: Please fill these gobo information

File: resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Beam-1500.qxf [hunk 4]
```diff
@@ -64,46 +64,40 @@
...
   <Capability Min="75" Max="92" Preset="GoboMacro" Res1="ClayPaky/gobo00014.svg">Gobo 4</Capability>
   <Capability Min="93" Max="111" Preset="GoboMacro" Res1="ClayPaky/gobo00013.svg">Gobo 5</Capability>
   <Capability Min="112" Max="129" Preset="GoboMacro" Res1="ClayPaky/gobo00012.svg">Gobo 6</Capability>
-  <Capability Min="130" Max="150">Gobo 1 Shake, Slow -&gt; Fast</Capability>
-  <Capability Min="151" Max="171">Gobo 2 Shake, Slow -&gt; Fast</Capability>
-  <Capability Min="172" Max="192">Gobo 3 Shake, Slow -&gt; Fast</Capability>
-  <Capability Min="193" Max="213">Gobo 4 Shake, Slow -&gt; Fast</Capability>
-  <Capability Min="214" Max="234">Gobo 5 Shake, Slow -&gt; Fast</Capability>
-  <Capability Min="235" Max="255">Gobo 6 Shake, Slow -&gt; Fast</Capability>
+  <Capability Min="130" Max="150" Preset="GoboShakeMacro" Res1="">Gobo 1 Shake, Slow -&gt; Fast</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Beam-1500.qxf:88]: Same as above

File: resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Spot-575-HPE.qxf [hunk 5]
```diff
@@ -93,28 +93,28 @@
...
   <Capability Min="56" Max="63" Preset="GoboMacro" Res1="ClayPaky/gobo00057.svg">Gobo 7</Capability>
   <Capability Min="64" Max="71" Preset="GoboMacro" Res1="ClayPaky/gobo00052.svg">Gobo 8</Capability>
-  <Capability Min="72" Max="113">Fast to slow rotation</Capability>
-  <Capability Min="114" Max="117">Stop rotation</Capability>
-  <Capability Min="118" Max="159">Slow to fast rotation</Capability>
-  <Capability Min="160" Max="255">Gobos shake</Capability>
+  <Capability Min="72" Max="113" Preset="RotationClockwiseFastToSlow">Fast to slow rotation</Capability>
+  <Capability Min="114" Max="117" Preset="RotationStop">Stop rotation</Capability>
+  <Capability Min="118" Max="159" Preset="RotationCounterClockwiseSlowToFast">Slow to fast rotation</Capability>
+  <Capability Min="160" Max="255" Preset="GoboShakeMacro" Res1="">Gobos shake</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Spot-575-HPE.qxf:99]: this is detailed in the manual
160-175 gobo 2
176-191 gobo 3
192-207 gobo 4
208-223 gobo 5
224-239 gobo 6
240-255 gobo 7

File: resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Spot-575-HPE.qxf [hunk 6]
```diff
@@ -93,28 +93,28 @@
...
-  <Capability Min="0" Max="17">No gobo</Capability>
+  <Capability Min="0" Max="17" Preset="GoboMacro" Res1="Other/Open.svg">No gobo</Capability>
   <Capability Min="18" Max="35" Preset="GoboMacro" Res1="ClayPaky/gobo00067.svg">Gobo 1</Capability>
   <Capability Min="36" Max="53" Preset="GoboMacro" Res1="ClayPaky/gobo00059.svg">Gobo 2</Capability>
   <Capability Min="54" Max="71" Preset="GoboMacro" Res1="ClayPaky/gobo00068.svg">Gobo 3</Capability>
   <Capability Min="72" Max="89" Preset="GoboMacro" Res1="ClayPaky/gobo00069.svg">Gobo 4</Capability>
   <Capability Min="90" Max="107" Preset="GoboMacro" Res1="ClayPaky/gobo00070.svg">Gobo 5</Capability>
   <Capability Min="108" Max="129" Preset="GoboMacro" Res1="ClayPaky/gobo00071.svg">Gobo 6</Capability>
-  <Capability Min="130" Max="255">Gobo shake</Capability>
+  <Capability Min="130" Max="255" Preset="GoboShakeMacro" Res1="">Gobo shake</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Alpha-Spot-575-HPE.qxf:110]: this is detailed in the manual
130-150 gobo 1
151-171 gobo 2
172-192 gobo 3
193-213 gobo 4
214-234 gobo 5
235-255 gobo 6

File: resources/fixtures/Clay_Paky/Clay-Paky-Mythos.qxf [hunk 7]
```diff
@@ -14,30 +14,30 @@
  <Channel Name="Yellow" Preset="IntensityYellow"/>
  <Channel Name="Colour 1">
   <Group Byte="0">Colour</Group>
-  <Capability Min="0" Max="27">Empty position</Capability>
-  <Capability Min="28" Max="49">Empty + Soft Filter</Capability>
+  <Capability Min="0" Max="27" Preset="ColorMacro" Res1="#ffffff">Empty position</Capability>
+  <Capability Min="28" Max="49" Preset="ColorDoubleMacro">Empty + Soft Filter</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Mythos.qxf:18]: please set the double colors on each entry

File: resources/fixtures/Clay_Paky/Clay-Paky-Mythos.qxf [hunk 8]
```diff
@@ -14,30 +14,30 @@
...
-  <Capability Min="235" Max="254">CTO 2500K + Blue Wood UV</Capability>
+  <Capability Min="235" Max="254" Preset="ColorDoubleMacro">CTO 2500K + Blue Wood UV</Capability>
   <Capability Min="255" Max="255" Preset="ColorMacro" Res1="#34378a">Blue Wood UV</Capability>
  </Channel>
  <Channel Name="Colour 2">
   <Group Byte="0">Colour</Group>
-  <Capability Min="0" Max="27">Empty position</Capability>
-  <Capability Min="28" Max="49">Empty + Dark Green</Capability>
+  <Capability Min="0" Max="27" Preset="ColorMacro" Res1="#ffffff">Empty position</Capability>
+  <Capability Min="28" Max="49" Preset="ColorDoubleMacro">Empty + Dark Green</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Mythos.qxf:32]: same as above

File: resources/fixtures/Clay_Paky/Clay-Paky-Mythos.qxf [hunk 9]
```diff
@@ -228,16 +218,16 @@
  </Channel>
  <Channel Name="Colour 3">
   <Group Byte="0">Colour</Group>
-  <Capability Min="0" Max="27">Empty position</Capability>
-  <Capability Min="28" Max="49">Empty + Light Green</Capability>
+  <Capability Min="0" Max="27" Preset="ColorMacro" Res1="#ffffff">Empty position</Capability>
+  <Capability Min="28" Max="49" Preset="ColorDoubleMacro">Empty + Light Green</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Mythos.qxf:222]: same as above

File: resources/fixtures/Clay_Paky/Clay-Paky-Sharpy.qxf [hunk 10]
```diff
@@ -102,16 +102,10 @@
...
-  <Capability Min="21" Max="41">Position 90°</Capability>
-  <Capability Min="42" Max="62">Position 180°</Capability>
-  <Capability Min="63" Max="83">Position 270°</Capability>
-  <Capability Min="84" Max="104">Position 360°</Capability>
-  <Capability Min="105" Max="126">Position 450°</Capability>
-  <Capability Min="127" Max="127">Position 540°</Capability>
-  <Capability Min="128" Max="190">Rotation (Fast 43rpm to slow 1.1rph)</Capability>
-  <Capability Min="191" Max="192">Stop</Capability>
-  <Capability Min="193" Max="255">Rotation (Slow 1.1rph to fast 43rpm)</Capability>
+  <Capability Min="0" Max="127" Preset="RotationIndexed">Rotation Indexed</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-Sharpy.qxf:105]: Why do you take the liberty to destroy a channel like this??
Revert

File: resources/fixtures/Clay_Paky/Clay-Paky-SuperSharpy.qxf [hunk 11]
```diff
@@ -14,30 +14,30 @@
  <Channel Name="Yellow" Preset="IntensityYellow"/>
  <Channel Name="Colour 1">
   <Group Byte="0">Colour</Group>
-  <Capability Min="0" Max="27">Empty position</Capability>
-  <Capability Min="28" Max="49">Empty + Soft Filter</Capability>
+  <Capability Min="0" Max="27" Preset="ColorMacro" Res1="#ffffff">Empty position</Capability>
+  <Capability Min="28" Max="49" Preset="ColorDoubleMacro">Empty + Soft Filter</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-SuperSharpy.qxf:18]: Again, if you set the macro, set also the colors, since you have them

File: resources/fixtures/Clay_Paky/Clay-Paky-SuperSharpy.qxf [hunk 12]
```diff
@@ -232,16 +222,16 @@
  </Channel>
  <Channel Name="Colour 3">
   <Group Byte="0">Colour</Group>
-  <Capability Min="0" Max="27">Empty position</Capability>
-  <Capability Min="28" Max="49">Empty + Light Green</Capability>
+  <Capability Min="0" Max="27" Preset="ColorMacro" Res1="#ffffff">Empty position</Capability>
+  <Capability Min="28" Max="49" Preset="ColorDoubleMacro">Empty + Light Green</Capability>
```
Comment (mcallegari) [inline resources/fixtures/Clay_Paky/Clay-Paky-SuperSharpy.qxf:226]: Same as above

**Review by mcallegari — COMMENTED** [review]
As usual, 5 fixtures reviewed, 10 errors.
Then I got bored.
I don't like the time I waste every time on your work.
At this point I could do it myself, without all those mistakes.

File: resources/fixtures/Clay_Paky/Clay-Paky-Sharpy.qxf (hunk 10 above)
Comment (joshuaurieli) [reply resources/fixtures/Clay_Paky/Clay-Paky-Sharpy.qxf:105]: The list of exact rotation positions (90°, 180°, 270°) for particular DMX values is a clumsy way to define the indexed rotation of a gobo. We have a preset for this: "RotationIndexed". Therefore we replaced the custom capabilities with a preset. We can revert this to being useless if you like.

**Comment by joshuaurieli** [comment]
@mcallegari Thank you for the review. I think we need to clarify the purpose of this pull request. We are **NOT** adding missing information to the fixtures from their respective manuals.  What we are **ARE** doing is adding gobo and rotation presets to **EXISTING** capabilities, so that the ActsOn property can be used. This is the same as in pull requests #1249 #1244 and was discussed extensively on the [forum](https://www.qlcplus.org/forum/viewtopic.php?f=35&t=14353). I have added comments to your comment that is not to do with adding new information.

File: resources/fixtures/Clay_Paky/Clay-Paky-Sharpy.qxf (hunk 10 above)
Comment (mcallegari) [reply resources/fixtures/Clay_Paky/Clay-Paky-Sharpy.qxf:105]: image

First, this is prism and not gobo. Second, this is not a "clumsy way". You intentionally fucked up a channel that was clearly defined in the product manual. See screenshot.

Comment (joshuaurieli) [reply resources/fixtures/Clay_Paky/Clay-Paky-Sharpy.qxf:105]: OK, our mistake. We will revert this and all similar changes. Please answer my [general comment blow](https://github.com/mcallegari/qlcplus/pull/1266#issuecomment-774985431).

**Comment by mcallegari** [comment]
The time I can spend on this project is very limited.
If I have a couple of hours, I prefer to spend them on bugfixing or advancing the QLC+5 developments so that I can finally release it.
In any case, this is my personal spare time. For you instead this is a job, and, I quote your website, "To avoid having to build another console-specific library or increase the cost of the console because a library must be rented, Ray Console uses QXF device templates". So basically you're using the QLC+ project efforts to make money with your product.
I can't even imagine how you parse QXF files in your code. You probably took QLC+ code (and maybe much more) and never gave credit to the QLC+ project (I can't find any reference in your website)

In conclusion, since we're both wasting time here with these back and forth, I don't intend to continue this "collaboration".
This is not how I wish to work on Pull Requests, especially since I tried many times to cooperate with you.
If you wish to submit new fixtures, you are welcome to do so, in the dedicated forum section, like any other QLC+ user.
I will review and eventually merge PR #1268 but I won't accept any other PR from you.
Nevertheless, thank you for the contribution you provided so far to this project, even with major difficulties.

**Comment by joshuaurieli** [comment]
Defining fixtures is a slow, laborious and unpleasant task for anyone building a DMX controller. We hoped that working with you would help us get started, help improve the QLC+ fixture library as well as establish a good open source fixture library for the benefit of all DMX controller projects.

Yes, we have benefited from the QLC+ fixture library, but we have also spent lots of time and energy contributing to it. We did not use QLC+ code, but parsed the QXF format ourselves. Credit to QLC+ is given in our software (along with all other open source code bases we employ) and we clearly state that we use the QLC+ fixture library wherever relevant.

Unfortunately the collaboration has also been extremely difficult from our side. We have tried hard to find workable compromises and become better at writing fixture definitions to your requirements. We will post any further QXF fixtures we create to the forum. Thank you for all time you have taken to discuss fixture definitions and review our pull requests. We wish you all the best with the development and release of QLC+ 5.
